source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,308 @@
+
+        const auto corners3D = estimate3DMarkerPoints(pointCloud, corners2D);
+        const auto centers3D = estimate3DMarkerPoints(pointCloud, centers2D);
+
+        auto origin = cv::Vec3d{ centers3D[0].x, centers3D[0].y, centers3D[0].z };
+        auto yAxis = cv::Vec3d{ corners3D[0].x - corners3D[1].x,
+                                corners3D[0].y - corners3D[1].y,
+                                corners3D[0].z - corners3D[1].z };
+
+        auto xAxis = cv::Vec3d{ corners3D[2].x - corners3D[1].x,
+                                corners3D[2].y - corners3D[1].y,
+                                corners3D[2].z - corners3D[1].z };
+
+        auto u = xAxis / cv::norm(xAxis, cv::NORM_L2);
+        auto v = yAxis / cv::norm(yAxis, cv::NORM_L2);
+        auto normal = u.cross(v);
+        auto unitNormal = normal / cv::norm(normal, cv::NORM_L2);
+
+        auto rotMatrix = cv::Mat(3, 3, CV_64F);
+        for(int i = 0; i < 3; ++i)
+        {
+            rotMatrix.at<double>(i, 0) = u[i];
+            rotMatrix.at<double>(i, 1) = v[i];
+            rotMatrix.at<double>(i, 2) = unitNormal[i];
+        }
+
+        auto transform = transformationMatrix(rotMatrix, origin);
+
+        if(display)
+        {
+            displayMarkerPose(pointCloud, corners2D, centers2D);
+        }
+
+        return transform;
+    }
+
+    Zivid::Matrix4x4 invertAffineTransformation(const Zivid::Matrix4x4 &transform)
+    {
+        auto rot = cv::Mat(3, 3, CV_32F);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                rot.at<float>(i, j) = transform(i, j);
+            }
+        }
+
+        auto trans = cv::Mat(3, 1, CV_32F);
+        for(int i = 0; i < 3; ++i)
+        {
+            trans.at<float>(i, 0) = transform(i, 3);
+        }
+
+        cv::Mat newRot = rot.inv();
+        cv::Mat newTrans = -newRot * trans;
+
+        auto cvInverted = cv::Mat(4, 4, CV_32F, 0.0);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                cvInverted.at<float>(i, j) = newRot.at<float>(i, j);
+            }
+        }
+        for(int i = 0; i < 3; ++i)
+        {
+            cvInverted.at<float>(i, 3) = newTrans.at<float>(i, 0);
+        }
+        cvInverted.at<float>(3, 3) = 1.0;
+
+        auto *ptr = &cvInverted.at<float>(0, 0);
+        auto invertTransform = Zivid::Matrix4x4(ptr, ptr + 16);
+
+        return invertTransform;
+    }
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        const auto dataFile = std::string(ZIVID_SAMPLE_DATA_DIR) + "/ArucoSample.zdf";
+        std::cout << "Reading ZDF frame from file: " << dataFile << std::endl;
+        const Zivid::Frame frame = Zivid::Frame(dataFile);
+        auto pointCloud = frame.pointCloud();
+
+        std::cout << "Converting from RGB to OpenCV BGR image" << std::endl;
+        cv::Mat image = pointCloudToBGR(pointCloud);
```

Comment:
'bgr' instead of 'image'? Isn't that what we use across samples?

Reply:
We actually use a few different things.
CaptureUndistortRGB -> cv::Mat imageToBGR()
CreateDepthMap and MaskPointCloud -> pointCloudToCvZ() and pointCloudToCvBGR()

I would either rename it to
        std::cout << "Converting to BGR image" << std::endl;
        cv::Mat image = pointCloudToCvBGR(pointCloud);
or
        std::cout << "Converting to OpenCV BGR image format" << std::endl;
        cv::Mat image = pointCloudToImage(pointCloud);

Reply:
I am not talking about the comment, but about the variable 'image'.
https://github.com/zivid/zivid-cpp-samples/blob/4fd1d27cf681f360a5ecd3ee220afc6a1170a1dd/source/Applications/Advanced/CaptureUndistortRGB/CaptureUndistortRGB.cpp#L67
https://github.com/zivid/zivid-cpp-samples/blob/4fd1d27cf681f360a5ecd3ee220afc6a1170a1dd/source/Applications/Advanced/CreateDepthMap/CreateDepthMap.cpp#L108

Reply:
> @chrisasc Have you introduced any // NOLINT that is not already used in the samples? If not sure, check please.

No, I removed those I added earlier. These are from the multi-camera samples that Torbjørn made

Reply:
ah, changed it to bgr